keps/sig-storage/1847-autoremove-statefulset-pvcs/README.md

```diff
@@ -0,0 +1,578 @@
+change the previously expected behaviour of existing Statefulset. 
+
+If the statefulset had been set with the RemoveOnStatefulSetDeletion 
+and RemoveOnScaleDown and the version of the kube-controller downgraded,
+even though the `PersistentVolumeClaimReclaimPolicy` field will go away, the references
+would still be acted upon by the garbage collector and cleaned up 
+based on the settings before downgrade. 
+
+<!--
+If applicable, how will the component be upgraded and downgraded? Make sure
+this is in the test plan.
+
+Consider the following in developing an upgrade/downgrade strategy for this
+enhancement:
+- What changes (in invocations, configurations, API use, etc.) is an existing
+  cluster required to make on upgrade in order to keep previous behavior?
+- What changes (in invocations, configurations, API use, etc.) is an existing
+  cluster required to make on upgrade in order to make use of the enhancement?
+-->
+
+### Version Skew Strategy
+There is only kubecontroller manager changes involved, hence not applicable for
+version skew involving other components.
+
+<!--
+If applicable, how will the component handle version skew with other
+components? What are the guarantees? Make sure this is in the test plan.
+
+Consider the following in developing a version skew strategy for this
+enhancement:
+- Does this enhancement involve coordinating behavior in the control plane and
+  in the kubelet? How does an n-2 kubelet without this feature available behave
+  when this feature is used?
+- Will any other components on the node change? For example, changes to CSI,
+  CRI or CNI may require updating that component before the kubelet.
+-->
+
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+* **How can this feature be enabled / disabled in a live cluster?**
+  - [x] Other
+    - The feature is enabled only if a user explicitly fills up the  `PersistentVolumeClaimReclaimPolicy`
+    - No special enablement or restart is required.
+
+* **Does enabling the feature change any default behavior?**
+  The default behaviour is only changed when user explicitly specifies the `PersistentVolumeClaimReclaimPolicy`. Hence no change in any user visible behaviour 
+  change by default.
+
+* **Can the feature be disabled once it has been enabled (i.e. can we roll back
+  the enablement)?**
+  Roll back is only possible by down grade. The side effects of downgrade are specified in the upgrade/downgrade section. But for Statefulset which is created with default settings has not impact on roll back.
+
+* **What happens if we reenable the feature if it was previously rolled back?** 
+For Statefulset PVCs which were previously set to a non-Retain `PersistentVolumeClaimReclaimPolicy` 
+the Statefulset is going to come back with VolumeClaimTemplates set to default, but 
+the experience would be that of non-retain policy - either get deleted on scale down or
```

Comment:
Should this be reconciling so that we remove any ownerrefs if they change the policy back? Although it will still be racy

Reply:
We're only adding the ownerrefs while a scaledown is in progress, so I think it will have to be reconciling given idempotency? Anytime we see desired replicas < actual replicas we'll set ownerrefs etc, which should regenerate any that have been manually removed.

Reply:
There's also the StatefulSetDelete policy where we add the ownerref on the statefulset object at creation time right?